ggml-sycl/common.hpp

```diff
@@ -294,5 +294,60 @@ struct ggml_backend_sycl_context {
     }
 };
 
+// common host functions
+
+static inline int get_work_group_size(const sycl::device& device) {
+    dpct::device_info prop;
+    dpct::get_device_info(prop, device);
+    return prop.get_max_work_group_size();
+}
+
+
+// common device functions
+
+static __dpct_inline__ float warp_reduce_sum(float x,
```

Comment:
can we separate ```warp_reduce_xxx``` to independent files?

Reply:
no, there are inline functions. there are only little codes in common.hpp.